server/semantics-service/src/main/java/edu/illinois/ncsa/incore/service/semantics/controllers/TypeController.java

```diff
@@ -104,17 +109,11 @@ public Response listTypes(
         @Parameter(name = "Limit number of results to return.") @DefaultValue("50") @QueryParam("limit") int limit,
         @Parameter(name = "List the hyperlinks.") @DefaultValue("false") @QueryParam("hyperlink") boolean hyperlink,
         @Parameter(name = "Return the full response.") @DefaultValue("false") @QueryParam("detail") boolean detail) {
-        Comparator<String> comparator = Comparator.naturalOrder();
-        if (order.equals("desc")) comparator = comparator.reversed();
 
-        List<Document> typeList = this.typeDAO.getTypes();
+        // import type comparator
+        Comparator<Type> comparator = typeComparator("name", order);
 
```

Comment:
Might as well add "name" to the parameter "sortBy"? similar to other controllers?

Reply:
@longshuicy I have updated the semanticsdb in dev. The 500 error was because there are a couple of types in prod/dev do not have column in tableSchema, which doesn't match the type model. I added a condition to prevent this exception.

Also, sortBy is a good catch! I also changed it too.

Reply:
Works well and thank you for the quick turnaround!